crates/node-maintainer/src/maintainer.rs

```diff
@@ -84,6 +91,9 @@ impl NodeMaintainerOptions {
         nm.run_resolver(self.kdl_lock).await?;
         #[cfg(debug_assertions)]
         nm.graph.validate()?;
```

Comment:
Now you might be wondering why don't validate after optimization? Unfortunately the edges in the optimized tree are not quite right. I do clone outgoing edges when duplicating nodes, but they point to original graph nodes and thus sometimes validation fails with "unreachable dependency" error.